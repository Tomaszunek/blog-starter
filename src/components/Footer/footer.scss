@import 'src/styles/variables.scss';

footer {
    background-color: $background;
    text-align: center;
    padding-bottom: 10px;
    border-top: 1px solid gray;
    .upperFooter {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 15px 0;
        margin-left: auto;
        margin-right: auto;
        width: 80%;
        
        .links {
            display: grid;
            grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr;
            grid-gap: 40px;
            margin-right: 120px;
            width: 45%;
    
            a {
                color: $font-c;
                font-size: 25px;
                text-decoration: none;

                &:hover {
                    color: rgb(233, 233, 233);
                    text-shadow: 1px 1px rgb(255, 255, 255);
                }

                &.linkedin {
                    background: url(/images/socialLogo/linkedin.png);
                    background-repeat: no-repeat;
                    background-size: 17%;
                    background-position: 0px -2px;
                    width: 160%;
                    &:hover {
                        color: #0077B5;
                        text-shadow: 1px 1px rgb(19, 140, 206);
                    }

                }
                &.facebook {
                    background: url(/images/socialLogo/fb.png);
                    background-repeat: no-repeat;
                    background-size: 17%;
                    background-position: 0px -2px;
                    width: 160%;
                    &:hover {
                        color: #3b5998;
                        text-shadow: 1px 1px #7286b1;
                    }
                }
                &.patronite {
                    background: url(/images/socialLogo/patreon.png);
                    background-repeat: no-repeat;
                    background-size: 15%;
                    background-position: 3px 1px;
                    width: 160%;
                    &:hover {
                        color: red;
                        text-shadow: 1px 1px rgb(255, 31, 31);
                    }
                }
                &.twitch {
                    background: url(/images/socialLogo/twitch.png);
                    background-repeat: no-repeat;
                    background-size: 16%;
                    background-position: 2px 1px;
                    width: 160%;
                    &:hover {
                        color: #6441a5;
                        text-shadow: 1px 1px #876bbb;
                    }
                }
                &.twitter {
                    background: url(/images/socialLogo/twitter.svg);
                    background-repeat: no-repeat;
                    background-size: 22%;
                    background-position: -5px -4px;
                    width: 160%;
                    &:hover {
                        color: #1da1f2;
                        text-shadow: 1px 1px #58b9f5;
                    }
                }
                &.youtube {
                    background: url(/images/socialLogo/youtube.png);
                    background-repeat: no-repeat;
                    background-size: 17%;
                    background-position: 0px -2px;
                    width: 160%;
                    &:hover {
                        color: #ff0000;
                        text-shadow: 1px 1px rgb(255, 31, 31);
                    }
                }
            }
        }

        form {
            display: grid;
            grid-gap: 5px;
            width: 25%;

            $breadcrumb-theme-1: rgba(20, 17, 219, 0.5);
            $breadcrumb-theme-2: rgba(255, 255, 255, 0.13);  
            $breadcrumb-theme-3: rgba(30, 67, 233, 0.5); 
            $breadcrumb-theme-4: rgba(33, 44, 196, 0.5); 

            input {
                background-color:  $breadcrumb-theme-1;
                color: white;
                border-radius: 8px;
                border: 3px solid $breadcrumb-theme-1;
                padding: 6px 8px;
                font-size: 18px;
                &::placeholder {
                    color: white;  
                }
                &:focus {
                    border-radius: 0px;
                    background-color: $breadcrumb-theme-3;
                }
            }
            textarea {
                background-color:  $breadcrumb-theme-1;
                color: white;
                border-radius: 8px;
                border: 3px solid $breadcrumb-theme-1;
                padding: 2px 8px;
                font-size: 18px;
                &::placeholder {
                    color: white;  
                }
                &:focus {
                    border-radius: 0px;
                    background-color: $breadcrumb-theme-3;
                }
            }

            button {
                background-color:  $breadcrumb-theme-1;
                color: white;
                border-radius: 8px;
                border: 3px solid $breadcrumb-theme-1;
                padding: 2px 8px;
                color: white;
                margin: 0 150px 0px 150px;
                font-size: 18px;
            }
        }
    }
    
    
    @media only screen and (min-width: $sm) and (max-width: $lg) {
        width: 100%;

        .upperFooter {
            margin-left: 0;
            margin-right: 0; 
            width: 75%;          

            .links {
                grid-template-columns: 1fr 1fr 1fr;
            }
        }
    }
    @media only screen and (min-width: $xs) and (max-width: $sm) {    
        width: 100%;

        .upperFooter {
            margin-top: 10px;   
            display: block;
            margin-left: auto;
            margin-right: auto;
            width: 80%;    

            .links {
                grid-template-columns: 1fr 1fr 1fr 1fr;
            }
            form {
                margin-top: 30px;                              
            }
        }
    }
    @media only screen and (max-width: $xs) {    
        width: 100%;

        .upperFooter {
            margin-top: 10px;   
            display: block;
            margin-left: auto;
            margin-right: auto;
            width: 80%;    

            .links {
                grid-template-columns: 1fr 1fr 1fr;
            }
            form {
                margin-top: 30px;                              
            }
        }
    }

}