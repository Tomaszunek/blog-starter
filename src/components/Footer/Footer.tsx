import * as React from 'react';
import './footer.scss'

export default class Footer extends React.Component {    
  public render() {
    return (
      <footer className="footer">
        <div className="upperFooter">
          <div className="links">
            {this.displayLinksBasedOnEnv(1)}          
            <a className="linkedin" href="">LinkedIn</a>
            <a className="patronite" href="">Patronite</a>
            <a className="twitter" href="">Twitter</a>
            {this.displayLinksBasedOnEnv(2)}
            <a className="facebook" href="">Facebook</a>
            <a className="twitch" href="">Twitch</a>
            <a className="youtube" href="">Youtube</a>
          </div>
          <form action="">
            <input type="text" placeholder="Who you are?"/>
            <textarea name="" id="" cols={30} rows={7} defaultValue="" placeholder="Please type your massage. :)"/>
            <button type="submit">Send</button>
          </form>          
        </div>
        <span>&copy; Created in 2018 by Tomasz Paw≈Çowski. All rights reserved.</span>
      </footer>
    );
  }

  private displayLinksBasedOnEnv = (num:number) => {
    if(process.env.serv_type === "blog") {
      if(num === 1) {
        return <>
            <a href="articles/tech">Technology</a>
            <a href="articles/spiritual">Spiritual</a>
            <a href="articles/psychology">Psychology</a>
        </>
      } else {
        return <>
            <a href="store">Store</a>
            <a href="articles/motivation">Motivation</a>
            <a href="login">Login</a>   
        </>
      }
    } else if(process.env.serv_type === "games") {
      if(num === 1) {
        return <>
            <a href="articles/news">News</a>
            <a href="projects">Projects</a>
            <a href="abilities">Abilities</a>   
        </>
      } else {
        return <>
            <a href="store">Store</a>
            <a href="about">About</a>
            <a href="login">Login</a>   
        </>
      }
    } else  {
      return <>
          <a href="">abc</a>
          <a href="">abc</a>
          <a href="">abc</a>   
      </>
    }
  }
}


