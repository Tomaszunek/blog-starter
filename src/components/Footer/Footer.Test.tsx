import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Footer from './Footer'

describe('Footer Page', () => {
  describe('Component should render right', () => {
    let app:any;             
    beforeEach(() => {
      app = mount(<Footer/>);
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<Footer/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })
    it('should contain links', () => {
      expect(app.find('a').length).toEqual(12);
    })
    it('should contain links containar', () => {
      expect(app.find('.links').length).toEqual(1);      
    })
    it('should contain links', () => {
      expect(app.find('.facebook').length).toEqual(1);
      expect(app.find('.linkedin').length).toEqual(1);
      expect(app.find('.patronite').length).toEqual(1);
      expect(app.find('.twitter').length).toEqual(1);
      expect(app.find('.twitch').length).toEqual(1);
      expect(app.find('.youtube').length).toEqual(1);
    })
    it('should contain copyright', () => {      
      expect(app.contains(<span>&copy; Created in 2018 by Tomasz Paw≈Çowski. All rights reserved.</span>)).toBeTruthy();
    })
    it('should contain contact form', () => {      
      expect(app.find('form').length).toEqual(1);
      expect(app.find('button[type="submit"]').length).toEqual(1);
    })
  });
}); 