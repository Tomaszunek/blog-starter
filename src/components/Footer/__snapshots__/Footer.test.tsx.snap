// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Page Component should render right render correctly snapshot component 1`] = `
<footer
  className="footer"
>
  <div
    className="links"
  >
    <a
      href=""
    >
      abc
    </a>
    <a
      href=""
    >
      abc
    </a>
    <a
      href=""
    >
      abc
    </a>
    <a
      className="linkedin"
      href=""
    >
      LinkedIn
    </a>
    <a
      className="patronite"
      href=""
    >
      Patronite
    </a>
    <a
      className="twitter"
      href=""
    >
      Twitter
    </a>
    <a
      href=""
    >
      abc
    </a>
    <a
      href=""
    >
      abc
    </a>
    <a
      href=""
    >
      abc
    </a>
    <a
      className="facebook"
      href=""
    >
      Facebook
    </a>
    <a
      className="twitch"
      href=""
    >
      Twitch
    </a>
    <a
      className="youtube"
      href=""
    >
      Youtube
    </a>
  </div>
  <form
    action=""
  >
    <input
      type="text"
    />
    <textarea
      cols={30}
      defaultValue=""
      id=""
      name=""
      rows={10}
    />
    <button
      type="submit"
    >
      Send
    </button>
  </form>
  <span>
    © Created in 2018 by Tomasz Pawłowski. All rights reserved.
  </span>
</footer>
`;
